### ISSUE TITLE
Perl-5.28.0-GCCcore-7.3.0.eb  not building due to perldoc sanity checks failing

### ISSUE THREAD

**Issue by willfurnass** [easybuilders/easybuild-easyblocks#1659]
```
$ eb --version
This is EasyBuild 3.8.1 (framework: 3.8.1, easyblocks: 3.8.1)
```

From [my build log](https://www.dropbox.com/s/0rmwuqpju9kfqiy/install_foss_2018b_toolchain_job.sge.o3588049.gz?dl=0) for Perl-5.28.0-GCCcore-7.3.0.eb the following looks like a major problem:
```
 /usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0/lib/perl5/5.28.0/xCouldn't copy cpan/podlators/blib/script/pod2man to /usr/local/scripts/pod2man: No such file or directory
```

I.e., it looks like it is trying to install stuff in the wrong place.

@boegel thinks the problem is that the Perl install process finds a `/usr/local/scripts` directory in my environment and incorrectly assumes that's where I'd like it to install scripts.  

[More background info](https://openpkg-dev.openpkg.narkive.com/bGejYSaD/bugdb-perl-possible-build-problem-copying-into-usr-local-scripts-pr-133) (from 16 years ago!)

Suggested fix: add the following to the Perl easyconfig (not tested yet):
```python
configopts = "-Dscriptdirexp=%(installdir)s/bin"
```

NB I've not yet had chance to test this!

**Comment by ocaisa** [comment]
Can you see if the easyblock update in #1660 fixes your problem? You can just download the raw [perl.py](https://raw.githubusercontent.com/easybuilders/easybuild-easyblocks/ed933f9a8f4bdce8696673101e1ef0a63091319a/easybuild/easyblocks/p/perl.py) and use `--include-easyblocks=/path/to/new/perl.py` in your `eb` command

**Comment by willfurnass** [comment]
@ocaisa Now getting build failures with:
```
$ eb --buildpath="${TMPDIR-/tmp}" --include-easyblocks=$PWD/perl.py --parallel=${NSLOTS-1} Perl-5.28.0-GCCcore-7.3.0.eb --robot
...
/bin/bash: -c: line 0: syntax error near unexpected token `('
== 2019-03-19 13:44:02,776 build_log.py:162 ERROR EasyBuild crashed with an error (at ?:124 in __init__): cmd " perl Makefile.PL PREFIX=/usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0  -Dscriptdirexp=%(installdir)s/bin" exited with exit code 1 and output:
/bin/bash: -c: line 0: syntax error near unexpected token `('
/bin/bash: -c: line 0: syntax error near unexpected token `('
/bin/bash: -c: line 0: syntax error near unexpected token `('
/bin/bash: -c: line 0: syntax error near unexpected token `('
== 2019-03-19 13:44:02,783 build_log.py:162 ERROR EasyBuild crashed with an error (at ?:124 in __init__): build failed (first 300 chars): cmd " perl Makefile.PL PREFIX=/usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0  -Dscriptdirexp=%(installdir)s/bin" exited with exit code 1 and output:
/bin/bash: -c: line 0: syntax error near unexpected token `('
== 2019-03-19 13:44:02,785 build_log.py:162 ERROR EasyBuild crashed with an error (at ?:124 in __init__): Build of /usr/local/community/rse/EasyBuild/sharc/easybuild/easyconfigs/p/Perl/Perl-5.28.0-GCCcore-7.3.0.eb failed (err: 'build failed (first 300 chars): cmd " perl Makefile.PL PREFIX=/usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0  -Dscriptdirexp=%(installdir)s/bin" exited with exit code 1 and output:\n/bin/bash: -c: line 0: syntax error near unexpected token `(\'\n/bin/bash: -c: line 0: ` perl Makefile.PL PREFIX=/usr/local/commun') (at easybuild/main.py:147 in build_and_install_software)
```

**Comment by ocaisa** [comment]
Ok, you've included
```configopts = "-Dscriptdirexp=%(installdir)s/bin"```
in your easyconfig, but with the easyblock change this should not be necessary.

**Comment by ocaisa** [comment]
...and doesn't seem to work anyway!

**Comment by willfurnass** [comment]
Ah, I'd forgotten I'd previously patched the easyconfig -  I've now reverted to using the 3.8.1 version.  I tried installing with `--include-easyblocks='/usr/local/community/rse/EasyBuild/notes/perl.py'` as a param to `eb`, where `perl.py` is the one you linked to.  Scripts not being put in the right place yet though.  

From the build log:

```
== 2019-03-19 17:20:05,659 run.py:141 INFO Auto-enabling streaming output of './Configure -de  -Dcc="gcc" -Dccflags="-O2 -ftree-vectorize -march=haswell -fno-math-errno -fPIC" -Dinc_version_list=none -Dscriptdirexp="/usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0"/bin -Dusethreads -Dprefix="/usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0"' command because logging to stdout is enabled
== (streaming) output for command './Configure -de  -Dcc="gcc" -Dccflags="-O2 -ftree-vectorize -march=haswell -fno-math-errno -fPIC" -Dinc_version_list=none -Dscriptdirexp="/usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0"/bin -Dusethreads -Dprefix="/usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0"':
== 2019-03-19 17:20:05,659 run.py:192 INFO running cmd: ./Configure -de  -Dcc="gcc" -Dccflags="-O2 -ftree-vectorize -march=haswell -fno-math-errno -fPIC" -Dinc_version_list=none -Dscriptdirexp="/usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0"/bin -Dusethreads -Dprefix="/usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0" 
== 2019-03-19 17:21:22,293 run.py:506 INFO cmd "./Configure -de  -Dcc="gcc" -Dccflags="-O2 -ftree-vectorize -march=haswell -fno-math-errno -fPIC" -Dinc_version_list=none -Dscriptdirexp="/usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0"/bin -Dusethreads -Dprefix="/usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0"" exited with exit code 0 and output:
== 2019-03-19 17:21:22,299 run.py:559 INFO parse_log_for_error msg: Command used: ./Configure -de  -Dcc="gcc" -Dccflags="-O2 -ftree-vectorize -march=haswell -fno-math-errno -fPIC" -Dinc_version_list=none -Dscriptdirexp="/usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0"/bin -Dusethreads -Dprefix="/usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0"
 ```

However, later on during the build:

```
...
/usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0/lib/perl5/5.28.0/xCouldn't copy cpan/podlators/blib/script/pod2man to /usr/local/scripts/pod2man: No such file or directory                                                                                                                                                                                         Couldn't chmod 755 /usr/local/scripts/pod2man: No such file or directory
...
== 2019-03-19 17:53:40,598 build_log.py:162 ERROR EasyBuild crashed with an error (at ?:124 in __init__): Sanity check failed: no file found at 'bin/perldoc' in /usr/local/community/rse/EasyBuild/software/Perl/5.28.0-GCCcore-7.3.0

```

I've added the following to the custom easyblock to see if it helps:
```
            '-Dscriptdir="{0}"/bin'.format(self.installdir),
            '-Dinstallscript="{0}"/bin'.format(self.installdir),
```

**Comment by willfurnass** [comment]
For info, I was able to install `Perl-5.28.0-GCCcore-7.3.0.eb`  after additionally setting both `scriptdir` and `installscript`.

**Comment by ocaisa** [comment]
I've included the other changes in #1660

**Comment by damianam** [comment]
Closing since the PR got merged
